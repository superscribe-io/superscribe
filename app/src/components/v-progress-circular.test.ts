import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VProgressCircular from './v-progress-circular.vue';

test('Mount component', () => {
	expect(VProgressCircular).toBeTruthy();

	const wrapper = mount(VProgressCircular, {
		slots: {
			default: 'Slot Content',
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('size props', () => {
	const props = ['x-small', 'small', 'large', 'x-large'];

	for (const prop of props) {
		const wrapper = mount(VProgressCircular, {
			props: {
				[prop]: true,
			},
		});

		expect(wrapper.classes()).toContain(prop);
	}
});

test('indeterminate prop', () => {
	const wrapper = mount(VProgressCircular, {
		props: {
			indeterminate: true,
		},
	});

	expect(wrapper.get('.circle').classes()).toContain('indeterminate');
});
