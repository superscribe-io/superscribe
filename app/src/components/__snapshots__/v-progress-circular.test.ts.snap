// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-0b5afddc=\\"\\" class=\\"v-progress-circular\\"><svg data-v-0b5afddc=\\"\\" class=\\"circle\\" viewBox=\\"0 0 30 30\\">
    <path data-v-0b5afddc=\\"\\" class=\\"circle-background\\" d=\\"M12.5,0A12.5,12.5,0,1,1,0,12.5,12.5,12.5,0,0,1,12.5,0Z\\" transform=\\"translate(2.5 2.5)\\"></path>
    <path data-v-0b5afddc=\\"\\" class=\\"circle-path\\" style=\\"stroke-dasharray: 0, 78.5;\\" d=\\"M12.5,0A12.5,12.5,0,1,1,0,12.5,12.5,12.5,0,0,1,12.5,0Z\\" transform=\\"translate(2.5 2.5)\\"></path>
  </svg>Slot Content</div>"
`;
